### PR TITLE
Update to use LibGit2Sharp v0.24

### LINKED ISSUE #1244: Errors opening a pull request: "Failed to nmap..."
Environment:
GitHub Enterprise
Git repo has two remotes. One is a fork.
VS 2017 15.3.5
GitHub.VisualStudio 2.3.3.38

Repro:
1) Open GitHub repo
2) Click Pull Requests
3) Select the fork remote
4) Select any pull request

Expected:
Opens PR

Actual:
Briefly spins and then displays the following error:

'Failed to nmap. No data written: Not enough storage is available to process this command.'

### PR DESCRIPTION
**Pull request by jcansdale**
This PR updates GHfVS to use the LibGit2Sharp v0.24 (the latest stable version).

> This is the last release before a moving to .NET Core compatible library.
>
> It will be the last supported release with the prior architecture; as a result, this release is primarily bugfixes and does not include major new APIs.

https://github.com/libgit2/libgit2sharp/releases/tag/v0.24

There are a number of point `#pragma warning disable 0618` statements, which allow us to use a few deprecated methods.

Unfortunately some of the suggested replacement methods expect a `Repository` object rather than an `IRepository`. This makes mocking little more awkward. Rather than refactor a bunch of our code and tests, I opted to simply disable and document the warnings.

The impetus to update was due the the following issue:

There have been reports (see #1244) of the following error on the GitHub pane:
`Failed to mmap. No data written: Not enough storage is available to process this command.`

It looks like this might be related to the version of `LibGit2` we're using. See comment by @ethomson here:
https://stackoverflow.com/a/38918860/121348
https://github.com/libgit2/libgit2/pull/3690

Fixes #1244

### CODE REVIEW

**Comment by jcansdale** [comment]
In this build, I've changed from using LibGit2Sharp 0.22 to 0.23.1 and from LibGit2Sharp.NativeBinaries.1.0.129 to 1.0.164.

[GitHub.VisualStudio-2.3.3.42-Debug.zip](https://github.com/github/VisualStudio/files/1359158/GitHub.VisualStudio-2.3.3.42-Debug.zip)

I'm going to ask the user if this fixes the issue they were seeing.

**Comment by ethomson** [comment]
👍  Let me know if it doesn't work and we can debug further.

**Comment by Cliff-Miller** [comment]
Hi everybody.
First I'd like to thanks to all support team. You're all awesome.

Unfortunately I have no more time to waiting for a solution. I reinstalled
VS2017 completely. All new and now, of course, GitHub works fine.

Thanks a lot indeed.

BR

2017-10-05 8:48 GMT-03:00 Jamie Cansdale <notifications@github.com>:

**Comment by jcansdale** [comment]
@ethomson Thanks for your offer of help debugging. 😄 

One user has tried a test build that updates from LibGit2Sharp 0.22 to 0.23.1 (and NativeBinaries.1.0.129 to 1.0.164). This initial report is encouraging. I'm awaiting feedback from another user.

**Comment by jcansdale** [comment]
Here's a build that includes this PR.
https://ghfvs-installer.github.com/releases/2.3.4.44/GitHub.VisualStudio.vsix

**Comment by meaghanlewis** [comment]
This looks good to me @jcansdale 👍 

Focused testing around fetching, pulling, pushing and checking out branches.

File: src/GitHub.Exports/Models/BranchModel.cs [hunk 1]
```diff
@@ -29,9 +29,11 @@ public BranchModel(LibGit2Sharp.Branch branch, IRepositoryModel repo)
         {
             Extensions.Guard.ArgumentNotNull(branch, nameof(branch));
             Extensions.Guard.ArgumentNotNull(repo, nameof(repo));
-
             Name = DisplayName = branch.FriendlyName;
-            Repository = branch.IsRemote ? new LocalRepositoryModel(branch.Remote.Url) : repo;
+#pragma warning disable 0618 // TODO: Replace `Branch.Remote` with `Repository.Network.Remotes[branch.RemoteName]`.
+            var remoteUrl = branch.Remote.Url;
```
Comment (jcansdale) [inline src/GitHub.Exports/Models/BranchModel.cs]: `Remote` is `null` if `IsRemote` is false.

**Pull request merged**